From 1cfcea124f8e9cdee1493a82ae990f169d2011bd Mon Sep 17 00:00:00 2001
From: tiemuhuaguo <tiemuhuaguo@tencent.com>
Date: Sun, 15 Oct 2023 10:35:30 +0800
Subject: [PATCH] build

---
 .gitignore     | 3 ++-
 CMakeLists.txt | 6 +++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index 706f7f8..c7f00bc 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,8 +1,9 @@
 /xcode/
-/build/
+/build*/
 /builds/
 /build-*/
 /_build/
+/install*/
 .DS_Store
 CMakeCache.txt
 DartConfiguration.tcl
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b57c853..de0784b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -594,7 +594,11 @@ if (REGISTER_BUILD_DIR)
   export (PACKAGE ${PACKAGE_NAME})
 endif ()
 if (REGISTER_INSTALL_PREFIX)
-  register_gflags_package(${CONFIG_INSTALL_DIR})
+  # register_gflags_package is very dangerous if you are using multi toolchains
+  # or the ld will try to link gflag.a compiled by ndk, when we are compiling glog of macos
+  # 不要把gflag-config.cmake的路径注册到~/.cmake/packages/gflags/，尤其是使用多个工具链编译的情况下
+  # 否则在编译macOS上的glog的时候，ld会试图链接ndk版本的gflag！！！！
+  # register_gflags_package(${CONFIG_INSTALL_DIR})
 endif ()
 configure_file (cmake/config.cmake.in "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}-config.cmake" @ONLY)
 
-- 
2.37.1 (Apple Git-137.1)

