From 70498ec339f2f59f0a58317749696cdc09c41ff4 Mon Sep 17 00:00:00 2001
From: tiemuhuaguo <tiemuhuaguo@tencent.com>
Date: Sun, 15 Oct 2023 10:43:33 +0800
Subject: [PATCH] build

---
 .gitignore     | 1 +
 CMakeLists.txt | 5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index 2678271..4b094a1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 *.orig
 /build*/
 bazel-*
+/install*/
\ No newline at end of file
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5c1e55..e9d647b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@ project (glog
   HOMEPAGE_URL https://github.com/google/glog
   LANGUAGES CXX
 )
+set(CMAKE_CXX_STANDARD 14)
 
 set (CPACK_PACKAGE_NAME glog)
 set (CPACK_PACKAGE_DESCRIPTION_SUMMARY "Google logging library")
@@ -81,7 +82,9 @@ if (WITH_GMOCK AND TARGET GTest::gmock)
 endif (WITH_GMOCK AND TARGET GTest::gmock)
 
 if (WITH_GFLAGS)
-  find_package (gflags 2.2.2)
+  find_package (
+    gflags 2.2.2 REQUIRED
+  )
 
   if (gflags_FOUND)
     set (HAVE_LIB_GFLAGS 1)
-- 
2.37.1 (Apple Git-137.1)

